src/plugins/blocksmith/components/imgur-post-block-modal/index.jsx

```diff
@@ -0,0 +1,207 @@
+    toggle: PropTypes.func.isRequired,
+    onAddBlock: PropTypes.func.isRequired,
+    onEditBlock: PropTypes.func.isRequired,
+  };
+
+  static defaultProps = {
+    isOpen: false,
+  };
+
+  constructor(props) {
+    super(props);
+
+    const { block } = props;
+
+    this.state = {
+      aside: block.get('aside'),
+      errorMsg: '',
+      hasUpdatedDate: block.has('updated_date'),
+      touched: false,
+      isValid: false,
+      imgurId: block.get('id'),
+      showContext: block.get('show_context'),
+      updatedDate: block.get('updated_date', new Date()),
+    };
+
+    this.handleAddBlock = this.handleAddBlock.bind(this);
+    this.handleChangeCheckbox = this.handleChangeCheckbox.bind(this);
+    this.handleChangeDate = this.handleChangeDate.bind(this);
+    this.handleChangeTextArea = this.handleChangeTextArea.bind(this);
+    this.handleChangeTime = this.handleChangeTime.bind(this);
+    this.handleEditBlock = this.handleEditBlock.bind(this);
+  }
+
+  componentDidMount() {
+    this.inputElement.focus();
+  }
+
+  setBlock() {
+    const {
+      hasUpdatedDate,
+      imgurId,
+      showContext,
+      updatedDate,
+      aside,
+    } = this.state;
+
+    const { block } = this.props;
+
+    return block.schema().createMessage()
+      .set('aside', aside)
+      .set('id', imgurId || null)
+      .set('show_context', showContext)
+      .set('updated_date', hasUpdatedDate ? updatedDate : null);
+  }
+
+  handleAddBlock() {
+    const { onAddBlock, toggle } = this.props;
+    onAddBlock(this.setBlock());
+    toggle();
+  }
+
+  handleEditBlock() {
+    const { onEditBlock, toggle } = this.props;
+    onEditBlock(this.setBlock());
+    toggle();
+  }
+
+  handleChangeDate(date) {
+    this.setState(changedDate(date));
+  }
+
+  handleChangeTime({ target: { value: time } }) {
+    this.setState(changedTime(time));
+  }
+
+  handleChangeCheckbox({ target: { id, checked } }) {
+    this.setState({ [id]: checked });
+  }
+
+  handleChangeTextArea(event) {
+    const input = event.target.value;
+    const id = getImgurPostMediaId(input);
+    let { errorMsg, imgurId, isValid } = this.state;
+
+    if (!id) {
+      errorMsg = 'url or embed code is invalid';
+      imgurId = input;
+    } else {
+      errorMsg = '';
+      imgurId = id;
+      isValid = true;
+    }
+    this.setState({
+      errorMsg,
+      imgurId,
+      isValid,
+      touched: true,
+    });
+  }
+
+  render() {
+    const {
+      aside,
+      errorMsg,
+      hasUpdatedDate,
+      showContext,
+      isValid,
+      touched,
+      imgurId,
+      updatedDate,
+    } = this.state;
+    const { isFreshBlock, isOpen, toggle } = this.props;
+
+    return (
+      <Modal isOpen={isOpen} toggle={toggle}>
+        <ModalHeader toggle={toggle}>{`${isFreshBlock ? 'Add' : 'Update'} Imgur Post Block`}</ModalHeader>
+        <ModalBody>
+          <FormGroup>
+            <Label>{isValid ? 'Post id' : 'Embed code, Link, URI, or Post id'}</Label>
```

Comment:
@jalopez13 uri is not an option with this one

Reply:
@efjacobson 

Removed URI from label.